syntax = "proto3";

package usermanagement;

// The User Management service definition.
service UserService {
  rpc CreateUser (CreateUserRequest) returns (CreateUserResponse);
  rpc GetUser (GetUserRequest) returns (GetUserResponse);
}

// The request message containing user details for creation.
message CreateUserRequest {
  string user_id = 1;
  string name = 2;
  string email = 3;
}

// The response message for user creation.
message CreateUserResponse {
  bool success = 1;
}

// The request message containing user ID for retrieval.
message GetUserRequest {
  string user_id = 1;
}

// The response message containing user details.
message GetUserResponse {
  string user_id = 1;
  string name = 2;
  string email = 3;
}

